### ISSUE TITLE
[C# Markup] ItemsRepeater crashes when used with ItemTemplateSelector

### ISSUE THREAD

**Issue by MartinZikmund** [unoplatform/uno.extensions#2407]
### Discussed in https://github.com/unoplatform/uno.extensions/discussions/2404

<div type='discussions-op-text'>

<sup>Originally posted by **kucint** July  1, 2024</sup>
`ItemTemplateSelector` works fine with `ListView`, but it crashes with `ItemsRepeater`:

```cs
public sealed partial class MainPage : Page
{
    public MainPage() => this
        .DataContext(new BindableMainPageModel(), (page, vm) => page
        .Background(ThemeResource.Get<Brush>("ApplicationPageBackgroundThemeBrush"))
        .Content(new StackPanel()
        .VerticalAlignment(VerticalAlignment.Center)
        .HorizontalAlignment(HorizontalAlignment.Center)
        .Children(
            new TextBlock()
            .Text("Hello Uno Platform!"),
            //new ListView()                             // ItemTemplateSelector works fine with ListView
            new ItemsRepeater()                          // but it crashes with ItemsRepeater
            .ItemsSource(() => vm.Items)
            .ItemTemplateSelector<Item>((item, selector) => selector
                .Default(() => new TextBlock().Text("def"))
                .Case<ItemA>(itemA => new TextBlock().Text("a"))
                .Case<ItemB>(itemB => new TextBlock().Text("b"))
        )));
}
```

on `net8.0-desktop`;
_An exception of type 'System.ArgumentException' occurred in Uno.UI.dll but was not handled in user code_

on `net8.0-windows10.0.19041`:
_An exception of type 'System.InvalidCastException' occurred in WinRT.Runtime.dll but was not handled in user code_

`"Uno.Sdk": "5.2.175"`

```
  <PropertyGroup>
    <UnoExtensionsVersion>4.1.24</UnoExtensionsVersion>
    <UnoToolkitVersion>6.0.18</UnoToolkitVersion>
    <UnoThemesVersion>5.0.13</UnoThemesVersion>
    <UnoCSharpMarkupVersion>5.2.14</UnoCSharpMarkupVersion>
  </PropertyGroup>
```

repro: [UnoTemplateSelectorApp.zip](https://github.com/user-attachments/files/16048288/UnoTemplateSelectorApp.zip)
</div>

**Comment by dansiegel** [comment]
@kucint this fix has been merged and should be available in the latest preview version. Please let me know if you have any further issues with this.

**Comment by kucint** [comment]
> @kucint this fix has been merged and should be available in the latest preview version. Please let me know if you have any further issues with this.
> 
> Also note that this will be backported to the 5.3 branch and will be in the upcoming 5.3 release.

Hi @dansiegel 
This fix must be something big, it seems that it fixed as well other issues related to `ItemsRepeater` I observed from time to time in my app but I could not find any stable repro pattern for them, so I did not report them. Now they all seem to be gone.
BTW, it has been already backported to 5.2, right?

Thanks for this fix, now I have impression that the implementation of `ItemsRepeater` is becoming stable!
Well done!
